### PR TITLE
Customize packages which OSTree will compose

### LINKED ISSUE #17: Customize Packages
We need to enable customizing (add/delete) packages in `fedora-atomic-docker-host.json`. And make sure that the user should be able to use own customized OSTree.

### PR DESCRIPTION
**Pull request by trishnaguha**
Fixes #17

### CODE REVIEW

File: compose.yml [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+
 - name: Clone Fedora-Atomic buildscripts
   git:
     repo=https://pagure.io/fedora-atomic.git
     version=f25
     depth=1
     dest=/workspace/fedora-atomic
 
+- name: Customize Packages
+  lineinfile:
```
Comment (gbraad) [inline compose.yml]: this works, but maybe writing to a new file... and composing this file would be a cleaner approach. WDYT?

Comment (trishnaguha) [reply compose.yml]: @gbraad Do you mean simply creating a copy of the json file and edit that? 
If you meant this, +1 from my side.

Comment (trishnaguha) [reply compose.yml]: We should also add it to `.gitignore` in that case.

Comment (gbraad) [reply compose.yml]: You would not create a copy, but a child: https://gitlab.com/gbraad/ceph-atomic/blob/downstream/ceph-atomic-host.json The key would be to use: `"include": "fedora-atomic-host.json",` to create the relationship.

Comment (trishnaguha) [reply compose.yml]: @gbraad Pushed the changes.

File: compose.yml [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
     depth=1
     dest=/workspace/fedora-atomic
 
+- name: Create JSON file for adding packages
+  file:
+    path: /workspace/fedora-atomic/customized-atomic-docker-host.json
+    state: touch
+
+- name:
+  lineinfile:
```
Comment (gbraad) [inline compose.yml]: can you use a ninja2 template for this instead?

**Review by gbraad — CHANGES_REQUESTED** [review]
use a template instead

**Comment by trishnaguha** [comment]
@gbraad It is composing the file `customized-atomic-docker-host.json` which includes `fedora-atomic-docker-host.json`.

File: compose.yml [hunk 3]
```diff
@@ -1,13 +1,28 @@
...
+    dest=/workspace/fedora-atomic/customized-atomic-docker-host.json
+    state=present
+    line='{\n    "include"{{':'}} "fedora-atomic-docker-host.json",\n    "packages"{{':'}} [\n        {{ packages }}\n    ]\n}'
```
Comment (trishnaguha) [inline compose.yml]: This is going to add the var `packages` in the json file created.

**Comment by gbraad** [comment]
* https://suspicious.website/2015/02/18/generating-config-files-with-ansible-and-jinja2/
  * https://blog.codecentric.de/en/2014/08/jinja2-better-ansible-playbooks-templates/

File: compose.yml (hunk 3 above)
Comment (trishnaguha) [reply compose.yml]: @gbraad Pushed with the requested changes. Can you review now?

File: templates/customized-packages-template.j2 [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+    "include":"fedora-atomic-docker-host.json",
```
Comment (gbraad) [inline templates/customized-packages-template.j2]: I would also consider this to be provided as a variable ;-)

File: compose.yml [hunk 5]
```diff
@@ -1,13 +1,30 @@
...
+    file: vars/packages.yml
+
 - name: Clone Fedora-Atomic buildscripts
   git:
     repo=https://pagure.io/fedora-atomic.git
     version=f25
     depth=1
     dest=/workspace/fedora-atomic
 
+- name: Create JSON file for adding packages
```
Comment (gbraad) [inline compose.yml]: Is this necessary?

File: compose.yml [hunk 6]
```diff
@@ -1,13 +1,30 @@
 ---
 # This playbook composes OSTree
+
+- include_vars:
+    file: vars/packages.yml
+
 - name: Clone Fedora-Atomic buildscripts
   git:
     repo=https://pagure.io/fedora-atomic.git
```
Comment (gbraad) [inline compose.yml]: possibly provide this as a variable too...

File: compose.yml [hunk 7]
```diff
@@ -1,13 +1,30 @@
...
     version=f25
```
Comment (gbraad) [inline compose.yml]: ditto

File: compose.yml [hunk 8]
```diff
@@ -1,13 +1,30 @@
...
     depth=1
     dest=/workspace/fedora-atomic
```
Comment (gbraad) [inline compose.yml]: ditto

File: compose.yml [hunk 9]
```diff
@@ -1,13 +1,30 @@
...
+  template:
+    src: ./templates/customized-packages-template.j2
+    dest: /workspace/fedora-atomic/customized-atomic-docker-host.json
+    owner: root
+    group: root
+    mode: 0700
+
 - name: Compose Fedora-Atomic Tree
   command:
     chdir=/workspace/fedora-atomic
```
Comment (gbraad) [inline compose.yml]: ditto

File: compose.yml [hunk 10]
```diff
@@ -1,13 +1,30 @@
...
 
+- name: Create JSON file for adding packages
+  file:
+    path: /workspace/fedora-atomic/customized-atomic-docker-host.json
+    state: touch
+
+- name: Template render for customized packages
+  template:
+    src: ./templates/customized-packages-template.j2
+    dest: /workspace/fedora-atomic/customized-atomic-docker-host.json
```
Comment (gbraad) [inline compose.yml]: part of a possible variable

**Review by gbraad — CHANGES_REQUESTED** [review]
Please consider extracting variables

**Comment by trishnaguha** [comment]
@gbraad Pushed the changes :-).

File: README.md [hunk 11]
```diff
@@ -28,7 +28,32 @@ Install Requirements - Start HTTP Server. After running the following playbook y
...
+```
+
+**Add the additional Packages you would like to have in the OSTree in `vars/buildrepo.yml`.**
+
+**If you wish to use CentOS Atomic modify the variables accordingly.**
+
+```
+---
+repo: https://pagure.io/fedora-atomic.git
+version: f25
```
Comment (gbraad) [inline README.md]: branch indicate it better

File: README.md [hunk 12]
```diff
@@ -28,7 +28,32 @@ Install Requirements - Start HTTP Server. After running the following playbook y
...
+repodir: fedora-atomic
```
Comment (gbraad) [inline README.md]: full path indicates more meaningful information and allows to be more flexible

File: README.md [hunk 13]
```diff
@@ -28,7 +28,32 @@ Install Requirements - Start HTTP Server. After running the following playbook y
...
+localcompose: customized-atomic-docker-host.json
```
Comment (gbraad) [inline README.md]: isn't the compose always local? so, what does the file really mean?

File: README.md [hunk 14]
```diff
@@ -28,7 +28,32 @@ Install Requirements - Start HTTP Server. After running the following playbook y
...
+sourcecompose: fedora-atomic-docker-host.json
```
Comment (gbraad) [inline README.md]: look at the documentation for the naming of these files.

File: README.md (hunk 13 above)
Comment (trishnaguha) [reply README.md]: Wasn't sure what to name it. I have named the json file cloned from source as `sourcecompose`.

**Comment by trishnaguha** [comment]
@gbraad Looks good now?

File: README.md [hunk 15]
```diff
@@ -28,7 +28,33 @@ Install Requirements - Start HTTP Server. After running the following playbook y
...
+
+**If you wish to use CentOS Atomic modify the variables accordingly.**
+
+```
+---
+repo: https://pagure.io/fedora-atomic.git
+branch: f25
+repodir: fedora-atomic
+abs_path: /workspace                                  # The absolute path to the git repo.
+coremanifest: customized-atomic-docker-host.json      # The manifest that goes into the custom host(ostree) content that we are going to build.
```
Comment (gbraad) [inline README.md]: core? why not custom?
Maybe I have it wrong, but core sounds off...

Comment (trishnaguha) [reply README.md]: We can name it `custommanifest`. IMO `coremanifest` sounds better since this is core manifest file we are using for the compose.

Comment (gbraad) [reply README.md]: but in this context it is a derived version. coremanifest can therefore be misunderstood for it's purpose.

Comment (trishnaguha) [reply README.md]: OK.

**Comment by gbraad** [comment]
Can you run this again on a clean install, and show me the output?

**Comment by trishnaguha** [comment]
@gbraad I have tested this on a new instance: 
Following is the output:

screenshot from 2017-03-04 13-40-46

**Review by gbraad — APPROVED** [review]

**Pull request merged**